### ISSUE TITLE
git with recursive=true always reports status as changed

### ISSUE THREAD

**Issue by LinusU** [ansible/ansible-modules-core#2094]
The git module always reports `changed` when run with `recursive=true`, even if there are no new commits at all.

It works great when running without `recursive=true` thought :+1:

**Comment by w1r0x** [comment]
Do you use track_submodules=yes? If so it can be related to #1509

**Comment by LinusU** [comment]
I do not use `track_submodules` so it's probably another issue...

**Comment by robinro** [comment]
@LinusU 
Can you provide a minimal working example to reproduce this? i.e. example playbook and repo?
`recursive=true` is not the only necessary condition (gives changed=false for me).

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by jctanner** [comment]
@LinusU any response to @robinro's question?

needs_info

**Comment by ansibot** [comment]
@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/30043
